proto/ptf/base_test.py

```diff
@@ -63,9 +60,7 @@ def stringify(n, length):
     'length' bytes, it is represented in the fewest number of bytes it
     does fit into without loss of precision.  It always returns a
     string at least one byte long, even if value=width=0."""
-    h = '%x' % n
-    s = ('0'*(len(h) % 2) + h).zfill(length*2).decode('hex')
-    return s
+    return int.to_bytes(n, length=length)
 
 def ipv4_to_binary(addr):
```

Comment:
I don't think the implementations of `ipv4_to_binary` and `mac_to_binary` are valid either

maybe a simple unit test file would be beneficial to validate these 3 functions, maybe using Python 3.8

Reply:
I've updated the pull request with unit tests for `stringify`, `ipv4_to_binary` and `mac_to_binary`.
The unit tests are running in `p4lang/third-party:latest` container that provides Python 3.8.